src/core/datatypes/trees/TopologyNode.cpp

```diff
@@ -2012,48 +1978,95 @@ void TopologyNode::setUseAges(bool tf, bool recursive)
 }
 
 
-void TopologyNode::setSampledAncestor(bool tf)
+/**
+ * If this node has an outdegree of 1 (i.e., only one descendant), either replace it
+ * by a bifurcation where one child is subtended by a zero-length branch, or
+ * remove it entirely, and then do the same for all its children by calling this function
+ * recursively.
+ */
+void TopologyNode::suppressOutdegreeOneNodes( bool replace )
 {
-    sampled_ancestor_tip = tf;
-
-    // Only tips can have the sampled_ancestor_tip flag set.
-    assert(not sampled_ancestor_tip or isTip());
-}
-
 
-void TopologyNode::setSpeciesName(std::string const &n)
-{
+    if ( isTip() == false )
```

Comment:
I don't see the point of this check when the next thing we do is checking the number of children